tests/test_queryselector.py

```diff
@@ -0,0 +1,301 @@
+    """Test ElementHandle.JJeval() should retrieve content from subtree."""
+    page = isolated_page
+    htmlContent = \
+        '<div class="a">not-a-child-div</div><div id="myId">' \
+        '<div class="a">a1-child-div</div><div class="a">a2-child-div</div></div>'
+    await page.setContent(htmlContent)
+    elementHandle = await page.J('#myId')
+    content = await elementHandle.JJeval('.a', "nodes => nodes.map(n => n.innerText)")
+    assert content == ['a1-child-div', 'a2-child-div']
+
+
+@chrome_only
+@sync
+async def test_ElementHandle_JJeval_returns_empty_list_if_no_selector(isolated_page):
+    """Test ElementHandle.JJeval() should return an empty list
+    and doesn't raises exception in case of missing selector.
+
+    """
+    page = isolated_page
+    htmlContent = '<div class="a">not-a-child-div</div><div id="myId"></div>'
+    await page.setContent(htmlContent)
+    elementHandle = await page.J('#myId')
+    nodesLength = await elementHandle.JJeval('.a', "nodes => nodes.length")
+    assert nodesLength == 0
+
+
+@chrome_only
+@sync
+async def test_ElementHandle_JJ_queries_elements(isolated_page):
+    """Test ElementHandle.JJ() should query existing elements."""
+    page = isolated_page
+    await page.setContent('<html><body><div>A</div><br/><div>B</div></body></html>')
+    html = await page.J('html')
+    elements = await html.JJ('div')
+    assert len(elements) == 2
+    assert [await page.evaluate("e => e.textContent", element) for element in elements] == ['A', 'B']
+
+
+@chrome_only
+@sync
+async def test_ElementHandle_JJ_returns_empty_list_if_no_element(isolated_page):
+    """Test ElementHandle.JJ() should return empty array for non-existing elements."""
+    page = isolated_page
+    await page.setContent('<html><body><span>A</span><br/><span>B</span></body></html>')
+    html = await page.J('html')
+    elements = await html.JJ('div')
+    assert not elements and elements == []
+
+
+@sync
+async def test_ElementHandle_Jx_queries_element(isolated_page, server):
+    """Test ElementHandle.Jx() should query existing element."""
+    page = isolated_page
+    await page.goto(server / 'playground.html')
+    await page.setContent('<html><body><div class="second"><div class="inner">A</div></div></body></html>')
+    html = await page.J('html')
+    second = await html.Jx("./body/div[contains(@class , 'second')]")
+    inner = await second[0].Jx("./div[contains(@class, 'inner')]")
+    content = await page.evaluate("e => e.textContent", inner[0])
+    assert content == 'A'
+
+
+@sync
+@chrome_only
+async def test_ElementHandle_Jx_returns_none_if_no_element(isolated_page):
+    """Test ElementHandle.Jx() should return None for non-existing element."""
+    page = isolated_page
+    await page.setContent('<html><body><div class="second"><div class="inner">B</div></div></body></html>')
+    html = await page.J('html')
+    second = await html.Jx("/div[contains(@class, 'third')]")
+    assert not second and second == []
```

Comment:
Again, the first check isn't needed:
```suggestion
    assert second == []
```

Reply:
Fixed